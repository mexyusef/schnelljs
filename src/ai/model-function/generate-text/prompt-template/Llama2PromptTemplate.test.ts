import { chat, instruction, text } from "./Llama2PromptTemplate.js";

describe("text prompt", () => {
  it("should format prompt", () => {
    const prompt = text().format("prompt");

    expect(prompt).toMatchSnapshot();
  });
});

describe("instruction prompt", () => {
  it("should format prompt with instruction", () => {
    const prompt = instruction().format({
      instruction: "instruction",
    });

    expect(prompt).toMatchSnapshot();
  });

  it("should format prompt with system and instruction", () => {
    const prompt = instruction().format({
      system: "system",
      instruction: "instruction",
    });

    expect(prompt).toMatchSnapshot();
  });

  it("should format prompt with instruction and response prefix", () => {
    const prompt = instruction().format({
      instruction: "instruction",
      responsePrefix: "response prefix",
    });

    expect(prompt).toMatchSnapshot();
  });
});

describe("chat prompt", () => {
  it("should format prompt with user message", () => {
    const prompt = chat().format({
      messages: [{ role: "user", content: "user message" }],
    });

    expect(prompt).toMatchSnapshot();
  });

  it("should format prompt with user-assistant-user messages", () => {
    const prompt = chat().format({
      messages: [
        { role: "user", content: "1st user message" },
        { role: "assistant", content: "assistant message" },
        { role: "user", content: "2nd user message" },
      ],
    });

    expect(prompt).toMatchSnapshot();
  });

  it("should format prompt with system message and user-assistant-user messages", () => {
    const prompt = chat().format({
      system: "you are a chatbot",
      messages: [
        { role: "user", content: "1st user message" },
        { role: "assistant", content: "assistant message" },
        { role: "user", content: "2nd user message" },
      ],
    });

    expect(prompt).toMatchSnapshot();
  });
});
