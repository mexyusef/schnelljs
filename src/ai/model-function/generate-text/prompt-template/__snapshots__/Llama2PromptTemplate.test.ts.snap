// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`chat prompt > should format prompt with system message and user-assistant-user messages 1`] = `
"<s>[INST] <<SYS>>
you are a chatbot
<</SYS>>

1st user message [/INST] assistant message </s><s>[INST] 2nd user message [/INST] "
`;

exports[`chat prompt > should format prompt with user message 1`] = `"<s>[INST] user message [/INST] "`;

exports[`chat prompt > should format prompt with user-assistant-user messages 1`] = `"<s>[INST] 1st user message [/INST] assistant message </s><s>[INST] 2nd user message [/INST] "`;

exports[`instruction prompt > should format prompt with instruction 1`] = `"<s>[INST] instruction [/INST] "`;

exports[`instruction prompt > should format prompt with instruction and response prefix 1`] = `"<s>[INST] instruction [/INST] response prefix"`;

exports[`instruction prompt > should format prompt with system and instruction 1`] = `
"<s>[INST] <<SYS>>
system
<</SYS>>

instruction [/INST] "
`;

exports[`text prompt > should format prompt 1`] = `"<s>[INST] prompt [/INST] "`;
